specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+                        webviewFrame.try_query<ICoreWebView2ExperimentalFrame>();
+                    if (frameExperimental)
+                    {
+                        frameExperimental->add_DOMContentLoaded(
+                            Callback<
+                                ICoreWebView2ExperimentalFrameDOMContentLoadedEventHandler>(
+                                [](
+                                    ICoreWebView2Frame* frame,
+                                    ICoreWebView2DOMContentLoadedEventArgs* args) -> HRESULT {
+                                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental;
+                                    frame->QueryInterface(IID_PPV_ARGS(&frameExperimental));
+
+                                    frameExperimental->ExecuteScript(
+                                        LR"~(
+                                        let content = document.createElement("h2");
+                                        content.style.color = 'blue';
+                                        content.textContent = "The url of the iframe is " + window.location.href.toString();
+                                        document.body.appendChild(content);
+                                        )~",
+                                        Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                            [](HRESULT error, PCWSTR result) -> HRESULT {
+                                                return S_OK;
+                                            })
+                                            .Get());
+                                    return S_OK;
+                                })
+                                .Get(),
+                            NULL);
+                    }
+                    return S_OK;
+                })
+                .Get(),
+            NULL));
+    }
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## IDL
+```c#
+/// WebView2Frame provides direct access to the iframes information and use.
+[uuid(423ba05f-55dd-4c50-8339-22afd009ed31), object, pointer_default(unique)]
+interface ICoreWebView2ExperimentalFrame : IUnknown {
+  /// Add an event handler for the `NavigationStarting` event.
+  /// A frame navigation will raise a `NavigationStarting` event and
+  /// a `CoreWebView2.FrameNavigationStarting` event. All of the
+  /// `FrameNavigationStarting` event handlers will be run before the
+  /// `NavigationStarting` event handlers. All of the event handlers share a
+  /// common `NavigationStartingEventArgs` object. Whichever event handler is
+  /// last to change the `NavigationStartingEventArgs.Cancel` property will
+  /// decide if the frame navigation will be cancelled. Redirects raise this
+  /// event as well, and the navigation id is the same as the original one.
+  ///
+  /// You may block corresponding navigations until the event handler returns.
```

Comment:
Not sure what this means?

Reply:
+1 Maybe it's trying to say "Navigations are not performed until the event handler returns"?

Reply:
"Navigations will be blocked until all (Frame)NavigationStarting event handlers return"